### PR TITLE
Add guidelines and coding conventions for contributors

### PR DESCRIPTION
**Pull request by uthaiyashankar**
## Purpose
This is to add the coding conventions and guidelines so that all developers participating in the project can follow similar practices.

### CODE REVIEW

**Review by dulvinw — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    For more information, go to [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+4. Accept the Contributor License Agreement (CLA)
+    
+    You need to Accept the Ballerina Contributor License Agreement (CLA) when prompted by a GitHub email notification after sending your first Pull Request (PR). Subsequent PRs will not require CLA acceptance.
+
+    If the CLA gets changed for some (unlikely) reason, you will be presented with the new CLA text after sending your first PR after the change.
+
+## Coding Conventions
+For nBallerina code, we are following [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html) with following changes. 
+1. Four space intent. Shouldn't use tab character. Most of the IDEs can be configured to replace tab character with 4 spaces.
```
Comment (ayonam) [inline CONTRIBUTING.md:51]: A small typo here.  "intent" should be "indent".

**Review by ayonam — APPROVED** [review]
Looks fine to me.  Good for merging.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    
+    You need to Accept the Ballerina Contributor License Agreement (CLA) when prompted by a GitHub email notification after sending your first Pull Request (PR). Subsequent PRs will not require CLA acceptance.
+
+    If the CLA gets changed for some (unlikely) reason, you will be presented with the new CLA text after sending your first PR after the change.
+
+## Coding Conventions
+For nBallerina code, we are following [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html) with following changes. 
+1. Four space indent. Shouldn't use tab character. Most of the IDEs can be configured to replace tab character with 4 spaces. 
+2. Write your code to fit within 120 columns.
+3. CamelCase naming conventions. Elaborate function names, class member names and global variable names.  Local variables can be short. Local variables and member variables should start with lowercase letter.
```
Comment (dulvinw) [inline CONTRIBUTING.md:53]:
```suggestion
3. CamelCase naming conventions. Elaborate function names, class member names and global variable names.  Local variables can be short. Local variables and member variables should start with lowercase letter. 
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (uthaiyashankar) [reply CONTRIBUTING.md]: Yes, thanks Ayonam. Fixed.

**Pull request merged**